test/server/cards/03_TWI/units/FavorableDelegate.spec.ts

```diff
@@ -0,0 +1,64 @@
+describe('Favorable Delegate', function () {
+    integration(function (contextRef) {
+        describe('Favorable Delegate\'s ability', function () {
+            describe('When played ability', function () {
+                beforeEach(function () {
+                    contextRef.setupTest({
+                        phase: 'action',
+                        player1: {
+                            hand: ['favorable-delegate']
+                        },
+                    });
+                });
+
+                it('should draw a card when played', function () {
+                    const { context } = contextRef;
+
+                    context.player1.clickCard(context.favorableDelegate);
+                    expect(context.player1.handSize).toBe(1);
+                });
+            });
+        });
+        describe('When defeated ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['royal-guard-attache'],
+                        groundArena: ['favorable-delegate']
+                    },
+                    player2: {
+                        leader: { card: 'mace-windu#vaapad-form-master', deployed: true },
+                        hasInitiative: true
+                    },
+                    autoSingleTarget: true
+                });
+            });
+
+            it('should discard a card', function () {
```

Comment:
This is another instance where we could consolidate tests a bit. We could join this into the previous test case by playing it from hand and then doing an attack with it. The third case could then just have its own separate setup

Reply:
I merged the draw and discard basic cases together, and kept the discard on empty deck. I could remove the beforeEach because of that